problems/arraysIntro.js

```diff
@@ -4,57 +4,86 @@
  * @param {string} fruit
  * @returns {string[]}
  */
-function addFruitToEnd() {}
+function addFruitToEnd (fruits, fruit) {
+  fruits.push(fruit);
+  return fruits;
+
+
+} 
+ 
 
 /**
  * Adds a fruit to the front of an array of fruits and returns the array.
  * @param {string[]} fruits
  * @param {string} fruit
  * @returns {string[]}
  */
-function addFruitToFront() {}
+function addFruitToFront (fruits, fruit) {
+  fruits.unshift(fruit)
+  return fruits;
+}
+
 
 /**
  * Removes the last fruit from an array of fruits and returns the array.
  * @param {string[]} fruits
  * @returns {string[]}
  */
-function removeLast() {}
+function removeLast(fruits) {
+  fruits.pop(fruits);
+  return fruits;
+}
 
 /**
  * Removes and returns the last fruit from from an array of fruits
  * @param {string[]} fruits
  * @returns {string} The fruit that was removed
  */
-function removeLastandReturnLast() {}
+function removeLastandReturnLast(fruits) {
+  let fruit = fruits.pop();
+  return fruit
+
+
+}
 
 /**
  * Takes in an array and returns the length
  * @param {Array} array
  * @returns {number}
  */
-function arrayLength() {}
+function arrayLength(array) {
+  return array.length;
+}
 
 /**
  * Removes the first fruit in an array of fruits and returns the array.
  * @param {string[]} fruits
  * @returns {string[]}
  */
-function removeFirst() {}
+function removeFirst (fruits) {
+  let fruit = fruits.shift();
```

Comment:
we want to avoid reassigning the value of the parameter so use a different variable name because if we need fruits again in the original state it will no longer exist